zebra/kernel_socket.c

```diff
@@ -1124,6 +1124,10 @@ void rtm_read(struct rt_msghdr *rtm)
 	} else
 		return;
 
+	if (rtm->rtm_type == RTM_GET || rtm->rtm_type == RTM_ADD)
+		if (rib_has_connected_route(&p, VRF_DEFAULT))
```

Comment:
This violates the dataplane <-> zebra boundary imo.  And I realize that we need to better layer this but I would prefer we don't make the problem worse.

Reply:
well ... for me that problem a showstopper, that why I did a patch 
right now we have:
- just skip all kernel routes with no IP gateway but with interface as gateway (my problem)
- all installed kernel routes has no ifindex set for nexthop on rib_add, which easily can create wrong kernel route in zebra in non-trivial cases (just possible inconsistency)
- connected routes on same interface may be duplicated (see comment above) (not touched by my patch)

So with the patch - I am adding kernel missed routes which are not mentioned already as `connected` to RIB. 
Not likely it may make things worse ...

Not sure that I understand what you meant under `we need to better layer this`

Reply:
> This violates the dataplane <-> zebra boundary

Can you explain what you meant under this violation?